### PR TITLE
Align the conditions under which benchmarks are run for different implementations

### PR DESCRIPTION
**Pull request by anmyachev**
Double check per request: https://github.com/intel/intel-xpu-backend-for-triton/pull/2186#issuecomment-2345147973

@whitneywhtsang regression confirmed: https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/10828800320/job/30044981178.

Why do we use [bfloat16](https://github.com/intel/intel-xpu-backend-for-triton/blob/80de30cbe482195e3220f272df3df49762f12e51/benchmarks/xetla_kernel/flash_attention/fmha_forward_v5.h#L26) type for XeTLA, but [float16](https://github.com/intel/intel-xpu-backend-for-triton/blob/80de30cbe482195e3220f272df3df49762f12e51/benchmarks/triton_kernels_benchmark/flash_attention_fwd_benchmark.py#L215) type for Triton and PyTorch? Should they be the same for a fair comparison?

Another interesting detail is that on 1100 GPU, regression is not reproduced.

Closes https://github.com/intel/intel-xpu-backend-for-triton/issues/2235

### CODE REVIEW

**Comment by whitneywhtsang** [comment]
> Why do we use [bfloat16](https://github.com/intel/intel-xpu-backend-for-triton/blob/80de30cbe482195e3220f272df3df49762f12e51/benchmarks/xetla_kernel/flash_attention/fmha_forward_v5.h#L26) type for XeTLA, but [float16](https://github.com/intel/intel-xpu-backend-for-triton/blob/80de30cbe482195e3220f272df3df49762f12e51/benchmarks/triton_kernels_benchmark/flash_attention_fwd_benchmark.py#L215) type for Triton and PyTorch? Should they be the same for a fair comparison?

Agree, we should use bfloat16 for Triton and PyTorch too for a fair comparison.

File: .github/workflows/triton-benchmarks.yml
Comment (anmyachev) [inline .github/workflows/triton-benchmarks.yml:1]: This change needs to be rolled back before merge, it was made to speed up testing in CI.

File: benchmarks/xetla_kernel/flash_attention/fmha_forward_v5.h [hunk 1]
```diff
@@ -644,10 +644,9 @@ sycl::event fmha_forward_impl(sycl::queue &q, void *_q, void *_k, void *_v,
   uint32_t size_ml = shape.get_ml_size();
 
   // forward
-  // T *query = sycl::malloc_shared<T>(size_query, q);
+  T *query = sycl::malloc_shared<T>(size_query, q);
```
[the remark this answers is not in the document]
Comment (whitneywhtsang) [reply benchmarks/xetla_kernel/flash_attention/fmha_forward_v5.h:647]: @anmyachev Thanks for the investigation! For a fair comparison, it makes sense to make the operating conditions as close as possible across implementations. 
Let's keep this PR up for a day so others can chime in if they have an opinion.

[the remark this answers is not in the document]
Comment (ESI-SYD) [reply benchmarks/xetla_kernel/flash_attention/fmha_forward_v5.h:647]: I think moving allocations out of kernel acceptable for behavior alignment.

Pls correct `sycl::ext::oneapi::bfloat16` to `sycl::half` in FA `XeTLA` kernel, I forget to modify back before,  they should have similar performance on this change

**Comment by anmyachev** [comment]
CI status: https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/10848473314

**Review by ESI-SYD — APPROVED** [review]

**Review by whitneywhtsang — APPROVED** [review]

**Pull request merged**